lms/util/lti_launch.py

```diff
@@ -11,55 +11,46 @@ def get_application_instance(session, consumer_key):
     ).one()
 
 
-def default_get_secret(request, consumer_key):
+def get_secret(request, consumer_key):
     """Retrieve the lti secret given."""
     instance = get_application_instance(request.db, consumer_key)
     return instance.shared_secret
 
 
-def default_get_lti_launch_params(request):
+def get_lti_launch_params(request):
     """Retrieve the lti launch params."""
     return dict(request.params)
 
 
-def lti_launch(get_lti_launch_params=default_get_lti_launch_params,
-               get_secret=default_get_secret):
+def lti_launch(view_function):
     """
-    Initialize decorator.
+    Handle the verification of an lms launch.
 
-    Allow caller to supply methods to customize extracting lti launch
-    params and retrieving lti secret
-    """
-    def decorator(view_function):
-        """
-        Handle the verification of an lms launch.
-
-        You should add this decorator before (logically) the route decorator.
-        For example:
-
-        @view_config(...)
-        @lti_launch
-        def some_view(request):
-        ...
-        """
-        def wrapper(request):
-            """Handle the lms validation."""
-            lti_params = get_lti_launch_params(request)
-            consumer_key = lti_params['oauth_consumer_key']
-            shared_secret = get_secret(request, consumer_key)
+    You should add this decorator before (logically) the route decorator.
```

Comment:
Oof :) Glad we're simplifying some of these; the ordering fragility scares me.

Reply:
Yeah there's a series of decorators here that for all practical intents and purposes are actually just one long function, just with the lines of code moved into four or five separate files